core/src/main/java/org/axonframework/commandhandling/gateway/DefaultCommandGateway.java

```diff
@@ -143,4 +122,48 @@ public DefaultCommandGateway(CommandBus commandBus, RetryScheduler retrySchedule
         send(command, new FailureLoggingCallback<>(logger, callback));
         return callback;
     }
+
+    /**
+     * Builder class to instantiate a {@link DefaultCommandGateway}.
+     * <p>
+     * The {@code dispatchInterceptors} are defaulted to an empty list.
+     * The {@link CommandBus} is a <b>hard requirements</b> and as such should be provided.
+     */
+    public static class Builder extends AbstractCommandGateway.Builder {
+
+        @Override
+        public Builder commandBus(CommandBus commandBus) {
+            super.commandBus(commandBus);
+            return this;
+        }
+
+        @Override
+        public Builder retryScheduler(RetryScheduler retryScheduler) {
+            super.retryScheduler(retryScheduler);
+            return this;
+        }
+
+        @Override
+        public Builder dispatchInterceptors(
+                MessageDispatchInterceptor<? super CommandMessage<?>>... dispatchInterceptors) {
+            super.dispatchInterceptors(dispatchInterceptors);
+            return this;
+        }
+
+        @Override
+        public Builder dispatchInterceptors(
+                List<MessageDispatchInterceptor<? super CommandMessage<?>>> dispatchInterceptors) {
+            super.dispatchInterceptors(dispatchInterceptors);
+            return this;
+        }
+
+        /**
+         * Initializes a {@link DefaultCommandGateway} as specified through this Builder.
+         *
+         * @return a {@link DefaultCommandGateway} as specified through this Builder
+         */
+        public DefaultCommandGateway build() {
+            return new DefaultCommandGateway(this);
+        }
```

Comment:
No validation method for the `CommandBus`?

As the Javadoc on the `Builder` class states "The {@link CommandBus} is a <b>hard requirements</b> and as such should be provided."

Reply:
The `AbstractCommandGateway#validate()` function already asserts the exact set this implementations wants asserted.
And as the `DefaultCommandGateway()` constructor calls the `AbstractCommandGateway()` constructor, which calls the `validate()` function, we should be safe.

Reply:
Ok great!